### PR TITLE
adding planning

### LINKED ISSUE #2: Planning
- [ ] Development Strategy
- [ ] Constraints
- [ ] Backlog
- [ ] Wireframe
- [ ] Retrospective
- [ ] README

### PR DESCRIPTION
**Pull request by sannesofie**
closes #2 

## Checklists

- [ ] All CI checks pass
- [ ] The branch works when you pull it and run it locally

### HTML

- [ ] the code is well-formatted
- [ ] the HTML code passes [validator.w3.org](https://validator.w3.org/)
- [ ] there are no inline styles (example: `style='color: red;'`)
- [ ] there are no `<style>` tags with CSS, all styles are hrefs
- [ ] there is no inline JavaScript (example: `onclick='doSomething()'`)
- [ ] there are no `<script>` tags with JS, all JS is in an separate file
- [ ] `id`s are used for JavaScript only, not for CSS
- [ ] semantic tags are used
- [ ] ARIA labels are used where necessary
- [ ] spelling and grammar is correct in all site content

### CSS

- [ ] the code is well-formatted
- [ ] the code does not use any `#` id's
- [ ] styles are responsive

### JavaScript

- [ ] code is clean and easy to read
- [ ] helpful variable names are used
- [ ] all comments are clear and help to understand the code
- [ ] there is no unused code in comments
- [ ] all file names are helpful and match their exports

#### src/data.js

- [ ] the file does not import anything
- [ ] the file only exports JS data (primitives, arrays, objects, ...)
- [ ] the file does not contain any functions or logic
- (optional) the data is documented with JSDocs

#### src/init

- [ ] there is an `index.js` that is included by the `index.html` file
- [ ] `../listener` files are imported
- [ ] there is no code that needs to run _after_ the page is initialized
- [ ] any other files have helpful names

#### src/listeners

- [ ] the DOM can be queried to find elements
- [ ] the DOM _is not_ be modified
- [ ] handlers are imported and used as callbacks to event listeners
- [ ] no other functions are imported or used
- [ ] the file name makes sense for the listener

#### src/handlers

- [ ] handlers have a JSDoc comment
- [ ] the function name matches the file name
- [ ] handlers are used as a callback to `.addEventListener` somewhere in the program
- [ ] functions from `/logic` or `/procedures` files may be imported and called
- [ ] functions from `/handlers` may be imported and attached to the DOM with event listeners
- [ ] data from `/data.js` may be imported and used
- [ ] handlers may read and write to the DOM
- [ ] handlers _do not_ return values that you will need later in the program

#### src/components

- [ ] A DOM element is returned
- [ ] handlers can be imported and used as callbacks to event listeners
- [ ] _logic_, _components_ and _procedures_ can also be imported
- [ ] there is an HTML test file to render the component with different inputs
- [ ] each component has a `.test.html` file
- [ ] bonus points for a unit test `.spec.js` file

#### src/logic

- [ ] the handler has a complete and correct JSDoc comment
- [ ] the function name matches the file name
- [ ] other functions from `/logic` may be imported and called
- [ ] the function _does not_ read or write to the DOM
- [ ] the function _does not_ log to the console
- [ ] each logic function has a `.spec.js` file
  - [ ] tests are simple with no extra code
  - [ ] tests are well-named
  - [ ] tests cover common use cases (bonus for edge cases)
  - [ ] tests check for side-effects (if necessary)

### CODE REVIEW

**Comment by sannesofie** [comment]
Hi @gelilaa this was a test. I did not expect it to pass because I did not touch the code yet. So if I understand it correctly we have to have the code fixed and be completely done with the whole planning before making the PR?

**Comment by gelilaa** [comment]
@sannesofie the test has to pass by the time you have fixed your logic function. so I suggest that you start with that and work your way up !!!

**Comment by sannesofie** [comment]
@gelilaa I tried everything but was not able to fix this lint problem ;(

**Comment by gelilaa** [comment]
> @gelilaa I tried everything but was not able to fix this lint problem ;(

I will comment on the places you have to fix

File: client/styles/index.css [hunk 1]
```diff
@@ -1 +1,113 @@
 @import url('./other-file.css');
+
+.animate-charcter {
+  text-transform: uppercase;
+  background-image: linear-gradient(
```
Comment (gelilaa) [inline client/styles/index.css]: 5:19  ✖  Expected newline after ":" with a multi-line declaration   declaration-colon-newline-after

File: client/styles/index.css [hunk 2]
```diff
@@ -1 +1,113 @@
...
+    -225deg,
+    #231557 0%,
+    #44107a 29%,
+    #ff1361 67%,
+    #fff800 100%
+  );
+  background-size: auto auto;
+  background-clip: border-box;
+  background-size: 200% auto;
```
Comment (gelilaa) [inline client/styles/index.css]: 14:3   ✖  Unexpected duplicate "background-size"                     declaration-block-no-duplicate-properties

File: client/styles/index.css [hunk 3]
```diff
@@ -1 +1,113 @@
...
+  color: #fff;
+  background-clip: text;
```
Comment (gelilaa) [inline client/styles/index.css]: 16:3   ✖  Unexpected duplicate "background-clip"                     declaration-block-no-duplicate-properties

File: client/styles/index.css [hunk 4]
```diff
@@ -1 +1,113 @@
...
+
+  -webkit-background-clip: text;
```
Comment (gelilaa) [inline client/styles/index.css]: 18:3   ✖  Unexpected empty line before declaration                   declaration-empty-line-before

File: client/styles/index.css [hunk 5]
```diff
@@ -1 +1,113 @@
...
+}
+
+.btn::before {
+  position: absolute;
+  content: '';
+  width: calc(100% + 4px);
+  height: calc(100% + 4px);
+  left: -2px;
+  top: -2px;
+  background: linear-gradient(
```
Comment (gelilaa) [inline client/styles/index.css]: 66:13  ✖  Expected newline after ":" with a multi-line declaration   declaration-colon-newline-after

File: client/styles/index.css [hunk 6]
```diff
@@ -1 +1,113 @@
...
+  left: 0;
+  top: 0;
+  border-radius: 10px;
+}
+
+@keyframes move {
+  0% {
+    background-position: 0 0;
+  }
+  50% {
```
Comment (gelilaa) [inline client/styles/index.css]: 107:3   ✖  Expected empty line before rule                            rule-empty-line-before

File: client/styles/index.css [hunk 7]
```diff
@@ -1 +1,113 @@
...
+    background-position: 400% 0;
+  }
+  100% {
```
Comment (gelilaa) [inline client/styles/index.css]: 110:3   ✖  Expected empty line before rule                            rule-empty-line-before

**Comment by sannesofie** [comment]
@gelilaa Surprisingly all checks have passed. Don't know how though, but happy to be able to deliver this ;)
